src/content/en/fundamentals/engage-and-retain/push-notifications/common-issues-and-reporting-bugs.md

```diff
@@ -0,0 +1,271 @@
+	"more_info": "http://autopush.readthedocs.io/en/latest/http.html\#error-codes",  
+	"message": "Request did not validate Invalid bearer token: Auth expired"  
+}
+```
+
+If the application server keys are different between when the user was
+subscribed and when the Authorization header was signed, a `Not Found`
+error will be returned:
+
+```
+{  
+	"errno": 102,  
+	"message": "Request did not validate invalid token",  
+	"code": 404,  
+	"more_info": "http://autopush.readthedocs.io/en/latest/http.html\#error-codes",  
+	"error": "Not Found"  
+}
+```
+
+Lastly if you have an invalid value in your JWT (for example if the "alg" value
+is an unexpected value) you'll receive the following error from Mozilla
+AutoPush:
+
+```
+{  
+	"code": 401,  
+	"errno": 109,  
+	"error": "Unauthorized",  
+	"more_info": "http://autopush.readthedocs.io/en/latest/http.html\#error-codes",  
+	"message": "Request did not validate Invalid Authorization Header"  
+}
+```
+
+## HTTP Status Codes
+
+There are a range of issues that can result in a non-201 response code from a
+push service. Below is a list of HTTP status codes and what it means in relation
+to web push.
+
+<table>
+<tr>
+<th>Status Code</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>429</td>
+<td>Too many requests. Your application server has reached a rate limit with a
+push service. The push service should include a 'Retry-After' header to
+indicate how long before another request can be made.</td>
+</tr>
+<tr>
+<td>400</td>
+<td>Invalid request. This generally means one of your headers is invalid or
+poorly formatted.</td>
+</tr>
+<tr>
+<td>404</td>
+<td>Not Found. This occurs when the subscription has expired. In this case you
+should delete the PushSubscription from your back end and wait for an
+opportunity to resubscribe the user.</td>
+</tr>
+<tr>
+<td>410</td>
+<td>Gone. The subscription is no longer valid and should be removed from your
+back end. This can be reproduced by calling \`unsubscribe()\` on a
+PushSubscription.</td>
+</tr>
+<tr>
+<td>413</td>
+<td>Payload size too large. The minimum size payload a push service must
+support is 4096 bytes (or 4kb), anything larger can result in this error.</td>
+</tr>
+</table>
+
+If the http status code is not in this list and the error message is not
+helpful, check the [Web Push Protocol
+spec](https://tools.ietf.org/html/draft-ietf-webpush-protocol) to see if the
+status code is referenced along with a scenario of when that status code can
+be used.
+
+## Payload Encryption Issue
+
+If you can successfully trigger a push message (i.e. send a message to a web
+push service and receive a 201 response code) but the push event never fires in
+your service worker, this is normally an indication that the browser failed to
+decrypt the message it received.
+
+If this is the case, you should see an error message in Firefox's DevTools
+console like so:
+
+![Firefox DevTools with Decryption Message](./images/ff-devtools-decryption-msg.png)
```

Comment:
Make sure your figure captions:
* are sentence case, but without an ending period
* match the capitalization of image text.
Please check these throughout.